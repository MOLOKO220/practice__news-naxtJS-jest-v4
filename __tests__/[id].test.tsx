import { render, screen, act } from "@testing-library/react";
import Id from "../pages/[id]";

global.fetch = jest.fn(() =>
  Promise.resolve({
    json: () =>
      Promise.resolve({
        by: "danieg",
        descendants: 101,
        id: 35627107,
        kids: [35628027, 35627475, 35627904, 35627413],
        score: 212,
        time: 1681905950,
        title:
          "Making a Linux home server sleep on idle and wake on demand – the simple way",
        type: "story",
        url: "https://dgross.ca/blog/linux-home-server-auto-sleep/",
      }),
  })
) as jest.Mock;

describe("CommentItem", () => {
  it("Fetch", async () => {
    await act(() => render(<Id id={343423534} />));

    expect(fetch).toHaveBeenCalledTimes(5);
    expect(
      screen.getByText(
        /Making a Linux home server sleep on idle and wake on demand – the simple way/i
      )
    ).toBeInTheDocument();
  });

  it("snapshot", () => {
    const { container } = render(<Id id={343423534} />);
    expect(container).toMatchSnapshot();
  });

  beforeEach(() => {
    jest.clearAllMocks();
  });
});
