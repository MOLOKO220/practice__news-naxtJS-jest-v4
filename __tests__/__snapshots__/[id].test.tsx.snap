// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentItem snapshot 1`] = `
<div>
  <main
    class="PagesOfNew"
  >
    <div>
      <h6 />
      <a
        class="PagesOfNew__url"
      />
      <div
        class="PagesOfNew__date"
      >
        NaN.NaN.NaN  NaN:NaN
      </div>
      <div
        class="PagesOfNew__author"
      >
        by: 
      </div>
      <div
        class="PagesOfNew__comments"
      >
        <p>
          Comments: 
        </p>
        <ul />
      </div>
    </div>
  </main>
</div>
`;
